agent/engine/docker_image_manager.go

```diff
@@ -273,6 +275,11 @@ func (imageManager *dockerImageManager) isImageOldEnough(imageState *image.Image
 	return ageOfImage > imageManager.minimumAgeBeforeDeletion
 }
 
+func (imageManager *dockerImageManager) nonECSImageOldEnough(NonECSImage ImageWithSizeID) bool {
+	ageOfImage := time.Now().Sub(NonECSImage.createdTime)
```

Comment:
why is this using "createdTime" instead of "PulledAt" time (that ecs images are using)? is there much difference between these?

Reply:
it should be `PulledAt`

Reply:
PulledAt may be null if the image was created locally -- I think thats the idea here.

Reply:
However, you may need to take both into consideration. if you go onto the host and 'docker pull busybox' its 'create' date may be days ago.

Reply:
hmm, this kinda breaks the logic of "old enough". an image could have been created a long time ago, but could have been used very recently.

Reply:
Do we want to track the created time or last used time of the image?

Reply:
Since from api response, we only know the the created time. Will change the env var to make more clear to let customer set the variable.

Reply:
I think the problem is right now it will remove regardless of any age.

Reply:
Can we check **PulledAt** first and if it is null, check the **createdTime**? Since docker does not rly have the feature to tell us last used time of image, we just assume **PulledAt** time for images that pulled and **createdTime** for images created on the host as last used time.

Reply:
@cyastella can you add some notes to the git commit message and PR summary explaining what problem we have now and what problem we are solving?

Reply:
yeah, will add it